### PR TITLE
[core] gRPC benchmark binary

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
Tools and simple instructions to do gRPC benchmark.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/rpc/test/grpc_bench/README [hunk 1]
```diff
@@ -0,0 +1,27 @@
+This package is used to benchmark gRPC's performance and overhead.
+
+To run the test, the instruction need to be followed in this:
+    https://github.com/Tradias/grpc_bench
+
+To run the test, a docker image need to be built and follow the instruction in grpc_bench to add new tests.
+
+-----------------------------------------------------------------------------------------------------------------------------------------
+| name                        |   req/s |   avg. latency |        90 % in |        95 % in |        99 % in | avg. cpu |   avg. memory |
```
Comment (rkooo567) [inline src/ray/rpc/test/grpc_bench/README:9]: how should we generate this output? Is it manual?

**Review by rkooo567 — APPROVED** [review]
So the conclusion from the benchmark is our implementation is pretty close to the raw grpc benchmark result, is this correct? Also is this test supposed to be running for all releases to catch regression, or one time is just fine?

**Comment by rkooo567** [comment]
I wonder if this could be a release test to detect grpc regressions

**Comment by iycheng** [comment]
@rkooo567 this only catches the server side work. Client work is not covered. (client work is done by the grpc_bench).

For now, I'll try to just make it run manually. I'll try to get a client so that we can run it in nightly.

File: src/ray/rpc/test/grpc_bench/README (hunk 1 above)
Comment (iycheng) [reply src/ray/rpc/test/grpc_bench/README:9]: Yea, you should follow the grpc_bench and add new test sute there. Not very trivial for now.

**Comment by iycheng** [comment]
Btw, if you check the cpu utilization, we are actually using more resources and still slightly slower than others.

**Pull request merged**